Vue app cleanup

Mojito-Fivem opened the pull request:
Got rid of the brunt of your bloat, reduced the bundle size by 38% by simply changing some dependencies.

restray commented on package.json:
```diff
@@ -17,7 +16,7 @@
         "jquery": "^3.6.0",
         "jquery-ui": "^1.13.1",
         "jsdom": "^19.0.0",
```
You can delete those lines too (don't need those, I used them for development)

restray reviewed (changes requested):
Just need to remove 3 lines in the `package.json` but you did a really good job! Thanks you so much :)

Mojito-Fivem commented:
I went through and deleted all instances of the `var` keyword, as of ES6 this is deprecated and let/const should be used in it's place. Also fixed some instances of using (==) instead of (===)

restray commented:
> If you're migrating your code to Vue you should at least consider using typescript here.

I don't agree with you at all... VueJs is already complicated for most of the servers owners! And most of the FiveM resources works with JS and not TS! Those are my arguments to "not moving on TS"

restray reviewed (approved):
It seems pretty good!

Mojito-Fivem commented:
> > If you're migrating your code to Vue you should at least consider using typescript here.
> 
> I don't agree with you at all... VueJs is already complicated for most of the servers owners! And most of the FiveM resources works with JS and not TS! Those are my arguments to "not moving on TS"

Exactly. If you're already going to put working on this resources out of the hands of 99% of the userbase, why would you not go the extra step and use typescript which gives a much better developer experience for not all that much work, I see you already using jsdocs here (some of them are wrong but this PR doesn't correct them) so why would you not simplify with typescript which will self document these functions

itschip commented on html_app/components/AttachmentMenu.vue:
```diff
@@ -183,19 +185,16 @@ export default {
-                    }
-                })
+            }).then(({data}) => {
+                if (data.AttachmentData !== null && data.AttachmentData !== undefined) {
```
You can simplify this with `!data.AttachmentData`

itschip commented on html_app/components/PlayerInventory.vue:
```diff
@@ -206,7 +206,7 @@ export default {
         },
         getInventoryItemAtSlot: function(inventory, slot) {
-            var selectedItem = this.items.filter((item) => item.inventory == inventory && item.slot == slot);
+            const selectedItem = this.items.filter((item) => item.inventory == inventory && item.slot == slot);
```
No strict equality here?

itschip reviewed (commented):
Something to look into.

Mojito-Fivem commented:
I'm not going to do it but I'm saying it should be considered if such a rewrite is being done in the first place.

xenfovn commented:
I think there will be a lot of people who need to see the number showing the total weight amount of the inventory for some purpose

Mojito-Fivem commented:
> I think there will be a lot of people who need to see the number showing the total weight amount of the inventory for some purpose

Relevant to this PR how?

IdrisDose commented:
Code looks good, has it been tested?

FjamZoo commented:
![image](https://user-images.githubusercontent.com/34508112/161474787-d757a182-af4b-4082-a75e-1edbb02b5bca.png)

I was getting this error which meant I couldn't open the inventory.

Mojito-Fivem commented:
In this state the inventory doesn't work as it is, this PR just serves to cleanup the poor practices in the code.

Mojito-Fivem commented:
fixed some occurrences of this pepega data.data usage but there's still issues in here of the css not being bundled by the webpack.

restray commented on html_app/components/PlayerInventory.vue:
```diff
@@ -389,28 +390,28 @@ export default {
 
             if (inventory == "give") {
-                axios.post("https://qb-inventory/GiveItem", {
+                fetchNui("GiveItem", {
```
Maybe strict equal over here ?

restray commented on html_app/components/PlayerInventory.vue:
```diff
@@ -335,35 +335,36 @@ export default {
             }
 
             this.amount = parseInt(this.amount);
+            let changeAmount
```
Maybe set the value to 0 ?

restray commented on html_app/index.js:
```diff
@@ -1,46 +1,42 @@
-        app.config.globalProperties.i18n = data.data
-    })
+fetchNui("RetrieveTranslations", {}).then((data) => {
+    app.config.globalProperties.i18n = data
```
Maybe add `data || {}` in case of error!

Mojito-Fivem replied:
I think that's beyond the scope of this PR.

restray replied:
You rewrite that part, that's why I'm asking you, I can make another PR after that ;)

restray reviewed (approved):
The code seems fine, all the comments I did, are just suggestions!
